BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransactionParser.java

```diff
@@ -23,49 +23,39 @@
 public class BunqTransactionParser implements TransactionParser {
 
     @Override
-    public void respondToPendingChallenges(Bank bank, PrivateKey privateKey, Collection<Transaction> transactionCollection) {
-        for (Transaction transaction: transactionCollection) {
-            String description = transaction.getDescription();
-            if (!isValidDescriptionFormat(description)) {
-                continue;
-            }
-            PublicKey publicKey = getPublicKeyForTransaction(transaction);
-            if (isValidChallenge(description, publicKey)) {
-                try {
-                    String response = createResponse(description, privateKey);
-                    sendTransaction(bank, transaction, response);
-                } catch (SignatureException | InvalidKeyException e) {
-                    Log.e("Transaction failed!", e.getLocalizedMessage());
-                }
-            }
-        }
-    }
-
-    private void sendTransaction(Bank bank, Transaction counterTransaction, String response) {
-        BunqTransaction transaction = new BunqTransaction(-0.01f, counterTransaction.getAccount(), counterTransaction.getCounterAccount(), Currency.getInstance("EUR"), response);
-        bank.sendTransaction(transaction);
+    public void respondToPendingChallenges(PrivateKey privateKey, Collection<Transaction> transactions, long cutoffTime) {
+        BankTeller teller = BankTeller.getBankTeller();
+        Blockchain chain = teller.getBlockchain();
+
+        transactions.stream()
+                .filter(t -> t.getValue() > 0)
+                .filter(t -> t.getDate().getTime() > cutoffTime)
+                .filter(t -> isValidDescriptionFormat(t.getDescription()))
+                .filter(t -> isValidChallenge(t.getDescription(), chain.getPublicKeyForIBAN(t.getCounterAccount().getIban())))
+                .forEach(t -> {
+                    try {
+                        teller.sendCent(t.getCounterAccount().getIban(),
```

Comment:
How is this dirty? I thought that the teller holds a bank object so you dont have to do the difficult setup yourself.